### PR TITLE
#118 Refactoring the filtering of the profiles (refactoring and enhancement)

### PR DESCRIPTION
**Pull request by MikaelKalajdzic**
This PR brings a better filtering of the profiles. A search bar now is always visible on top of the profiles list. The user can now open a dialog for advanced profiles filtering in the community activity.

### CODE REVIEW

File: app/src/main/java/com/sdpteam/connectout/profileList/FilteredProfileListFragment.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        this(ProfileFilter.NONE);
+    }
+
+    public FilteredProfileListFragment(ProfileFilter baseFilter) {
+        this.baseFilter = baseFilter;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View contentView = inflater.inflate(R.layout.fragment_filter, container, false);
```
Comment (DavidKalajdzic) [inline app/src/main/java/com/sdpteam/connectout/profileList/FilteredProfileListFragment.java:35]: But here there could be an error as well (not sure at all)
From what was done previously (see file that you deleted in this PR : EventParticipantsListFragment for instance) we provide here the layout which is the xml file name not a xml child like here.

**Review by DavidKalajdzic — COMMENTED** [review]
Thank you for this work, the refactor looks difficult.
But unfortunately I cannot approve it now because it does not work, I tried to find bug/direction ideas for solving this, take a look at my comments

File: app/src/main/java/com/sdpteam/connectout/profileList/FilteredProfileListFragment.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        final ProfilesViewModel viewModel = new ProfilesViewModel(new ProfileFirebaseDataSource());
+        viewModel.setFilter(baseFilter);
+
+        final EditText searchFilter = contentView.findViewById(R.id.text_filter);
+
+        searchRunnable = () -> {
+            final ProfileNameFilter nameFilter = new ProfileNameFilter(searchFilter.getText().toString());
+            viewModel.setFilter(baseFilter.and(nameFilter));
```
[the remark this answers is not in the document]
Comment (MikaelKalajdzic) [reply app/src/main/java/com/sdpteam/connectout/profileList/FilteredProfileListFragment.java]: Yes certainly, but then I can't combine the filters :(

File: app/src/main/java/com/sdpteam/connectout/profileList/FilteredProfileListFragment.java (hunk 1 above)
Comment (MikaelKalajdzic) [reply app/src/main/java/com/sdpteam/connectout/profileList/FilteredProfileListFragment.java:35]: For now it works, we can investigate further I agree

**Review by DavidKalajdzic — APPROVED** [review]
It works thanks!

**Pull request merged**